redwood-treehouse-host/src/commonTest/kotlin/app/cash/redwood/treehouse/FakeCodeSession.kt

```diff
@@ -18,22 +18,32 @@ package app.cash.redwood.treehouse
 import app.cash.redwood.treehouse.CodeSession.Listener
 import app.cash.redwood.treehouse.CodeSession.ServiceScope
 import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.job
+import kotlinx.coroutines.launch
 import kotlinx.serialization.json.Json
 
 internal class FakeCodeSession(
```

Comment:
I think I can refactor the ZiplineCodeSession state management to share a lot more code with this fake.